### PR TITLE
+ Remote debugging for Filesystem workspace

### PR DESCRIPTION
**Pull request by iHardRock**
Remote debugging now available in FS workspace as well.

изображение

Also bug with Program path on General  tab in FS workspace fixed. It was may be self-cleared after doing some changes on other tabs and then save not returning to General tab.

### CODE REVIEW

**Comment by iHardRock** [comment]
@eranif, please check this PR :)

File: Plugin/FileSystemWorkspace/FSConfigPage.cpp [hunk 1]
```diff
@@ -156,7 +164,7 @@ void FSConfigPage::Save()
         last_executables.swap(small_arr);
     }
 
-    m_config->SetExecutable(m_comboBoxExecutable->GetStringSelection());
+    m_config->SetExecutable(m_comboBoxExecutable->GetValue()); //->GetStringSelection());
```
Comment (Jarod42) [inline Plugin/FileSystemWorkspace/FSConfigPage.cpp:167]: I would remove commented code.
Can be retrieved from history anyway.

File: Plugin/FileSystemWorkspace/clFileSystemWorkspace.cpp [hunk 2]
```diff
@@ -999,8 +1005,30 @@ void clFileSystemWorkspace::OnDebug(clDebugEvent& event)
...
+    
+    if (dbgr->GetIsRemoteDebugging()) {
+        // debugging remote target
+        wxString comm;
+        wxString host = GetConfig()->GetDebuggerRemoteHost();
+        wxString port = GetConfig()->GetDebuggerRemotePort();
+        
+        comm << host;
+
+        host = host.Trim().Trim(false);
```
Comment (Jarod42) [inline Plugin/FileSystemWorkspace/clFileSystemWorkspace.cpp:1017]: You probably invert those 2 lines. else modifying host is useless.

Comment (iHardRock) [reply Plugin/FileSystemWorkspace/clFileSystemWorkspace.cpp:1017]: Yes, you right. It was blindly copied from LiteEditor/manager.cpp (around line 2238) :)
So this "bug" already present in CodeLite (https://github.com/eranif/codelite/blob/287ff78fa959f6f4491dea73ea1dc5bf92710cd3/LiteEditor/manager.cpp#L2080)

**Review by Jarod42 — APPROVED** [review]

**Comment by eranif** [comment]
> @eranif, please check this PR :)

Sorry for the slow response time, been a busy week at work. Will look into it this weekend

**Pull request merged**